### PR TITLE
DBT-767: Improve the performance of functional tests.

### PR DESCRIPTION
**Pull request by vamshikolanu**
## Describe your changes
As part of this PR, Made changes to improve the performance of functional tests. The previous runtime for all the tests was around 90 mins. Now all the tests complete under 12 mins. Was able to achieve this by running tests in parallel using pytest-xdist library. 

Also using this test suite, we can test the adapter for multiple versions of python using tox. 

## Internal Jira ticket number or external issue link:
https://jira.cloudera.com/browse/DBT-767

## Testing procedure/screenshots(if appropriate):
To Run an individual test using tox:  https://gist.github.com/vamshikolanu/f97dfde28774f5af14d6d2cce6258daa
To Run an individual test using tox for different python version:  https://gist.github.com/vamshikolanu/2226b4fc343c8c1191fb1e1b263da5ca

To Run all tests using tox: https://gist.github.com/vamshikolanu/f97dfde28774f5af14d6d2cce6258daa
To Run all tests across all python versions installed on the machine: https://gist.github.com/vamshikolanu/7970c5cf131a6e9a5f4a85795c73f02a

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have formatted my added/modified code to follow pep-8 standards
- [x] I have checked suggestions from python linter to make sure code is of good quality.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,46 +1,20 @@
...
-$(VENV)/bin/activate:
-	@\
-	python3 -m venv $(VENV)
-
-install_deps: $(VENV)/bin/activate dev-requirements.txt	## Install all dependencies.
-	@\
-	$(VENV)/bin/python3 -m pip install --upgrade pip
-	$(VENV)/bin/pip install -e . -r dev-requirements.txt
-
-dev_setup: $(VENV)/bin/activate	 ## Install all dependencies and setup pre-commit.
```
Comment (niteshy) [inline Makefile:27]: @vamshikolanu why we need to remove the venv environment setup? 
How do we install python modules? Earlier the make dev_setup ensure the setup of venv and install required python modules. We need to manually install the dependencies like nox too !!

File: noxfile.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import nox
+
+package = "dbt-hive"
+python_versions = ["3.7", "3.8", "3.9", "3.10", "3.11"]
```
Comment (niteshy) [inline noxfile.py:4]: How do we ensure that all these python versions are installed in the machine?

File: noxfile.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+dbt_versions = ["1.5.0"]
```
Comment (niteshy) [inline noxfile.py:5]: Why do we need to hard-code the dbt-version? This will become one more place to update the dbt-version going forward!

Comment (vamshikolanu) [reply noxfile.py]: Removed the dependency of dbt version.

File: noxfile.py (hunk 2 above)
Comment (vamshikolanu) [reply noxfile.py]: We can't. Users need to install these versions of python on their machine. Tox throws a warning if the version is not present.

File: Makefile (hunk 1 above)
Comment (vamshikolanu) [reply Makefile]: Removed dependency of nox and reverted dev_setup steps.

File: Makefile [hunk 4]
```diff
@@ -29,11 +28,11 @@ dev_setup: $(VENV)/bin/activate	 ## Install all dependencies and setup pre-commi
 	make install_deps
 	$(VENV)/bin/pre-commit install
 
-functional_test:	## Run functional tests.
-	$(VENV)/bin/python3 -m pytest --profile $(PROFILE) $(TESTS) $(FLAGS)
+test: ## Test specific version of python
+	tox --recreate -e $(PYTHON_VERSION) -- $(TESTS) --profile $(PROFILE)
```
Comment (niteshy) [inline Makefile:32]: tox won't be able as a top-level command in the shell. Either add it to the shell or provide the path to tox. 

```
make test TESTS=tests/functional/adapter/test_basic.py::TestSimpleMaterializationsHive PYTHON_VERSION=py38
tox --recreate -e py38 -- tests/functional/adapter/test_basic.py::TestSimpleMaterializationsHive --profile dwx_endpoint
make: tox: Command not found
make: *** [Makefile:32: test] Error 127

```

Comment (vamshikolanu) [reply Makefile]: Thanks for catching this. Updated the PR.

**Review by niteshy — APPROVED** [review]

**Pull request merged**